### DISCUSSION TITLE
JSON API

### DISCUSSION THREAD

**Discussion by Summertime** [comfies/circle#6] [Ideas]
via @mininmobile 
- - -

### `/api/list.json`
Returns the full list of domain ids.
```json
[
    "comfi.es",
    "dystopia.club",
    "zvava.org",
    "lllil.li"
    // etc...
]
```

### `/api/sites.json`
Returns the full list of site objects.
```json
[
    { "id": "comfi.es", ... },
    { "id": "dystopia.club", ... },
    { "id": "zvava.org", ... },
    { "id": "lllil.li", ... }
    // etc...
]
```

### `/api/domain.json`
Returns info about a domain id.
```json
{
    "id": "domain",                       // domain's id
    "url": "https://domain/",             // domain's full url
    "css": "https://domain/webcircle.css" // domain's css (if any)
}
```

### `/api/f/domain.json`
Returns the previous and next site objects in the circle along with some other info, useful for constructing custom embeds.
```json
{
    "home": "https://circle.comfi.es/",    // home of webcircle
    "css": "https://domain/webcircle.css", // domain's css (if any)
    // previous website in the circle
    "prev": {
        "id": "prev-domain",
        "url": "https://prev-domain/",
        "css": "https://prev-domain/webcircle.css",
    },
    // next website in the circle
    "next": {
        "id": "next-domain",
        "url": "https://next-domain/",
        "css": "https://next-domain/webcircle.css",
    }
}
```

**Comment by Summertime** [comment]
I think we could have `/api/sites.json` just return a list of `id`s and `url`s, (the previous/next aspect is covered by the ordered list, and CSS isn't much useful in a API context)

for `/api/example.com.json` (la `/api/f/domain.json`), once again remove the `css` vars, and the home var (can be deduced by just going to `/` on origin)

and I don't think we'd need the other two endpoints

**Reply by mininmobile** [reply]
hmm yes okay i agree anyways what the fuck is this github feature what is going on?
